service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/dataproccluster/DeleteDataprocClusterStep.java

```diff
@@ -0,0 +1,81 @@
+package bio.terra.workspace.service.resource.controlled.cloud.gcp.dataproccluster;
+
+import bio.terra.cloudres.google.dataproc.ClusterName;
+import bio.terra.cloudres.google.dataproc.DataprocCow;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.utils.GcpUtils;
+import bio.terra.workspace.service.crl.CrlService;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.dataproc.model.Operation;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Optional;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** A step for deleting a controlled Dataproc cluster. */
+public class DeleteDataprocClusterStep implements Step {
+  private final Logger logger = LoggerFactory.getLogger(DeleteDataprocClusterStep.class);
+
+  private final ControlledDataprocClusterResource resource;
+  private final CrlService crlService;
+
+  public DeleteDataprocClusterStep(
+      ControlledDataprocClusterResource resource, CrlService crlService) {
+    this.resource = resource;
+    this.crlService = crlService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    ClusterName clusterName = resource.toClusterName();
+    DataprocCow dataprocCow = crlService.getDataprocCow();
+    try {
+      Optional<Operation> rawOperation = deleteIfFound(clusterName, dataprocCow);
```

Comment:
nit: this logic flow was a little hard for me to follow. I wonder if it would be simpler to skip the subroutine?